src/main/kotlin/com/sourcegraph/utils/CodyFormatter.kt

```diff
@@ -41,13 +42,14 @@ class CodyFormatter {
         // Fix for the IJ formatting bug which removes spaces even before the given formatting
         // range.
         val existingStart = appendedString.substring(0, cursor)
-        val formattedStart = psiFile.text.substring(0, cursor)
+        val endIndexInPsiFile = min(cursor, psiFile.text.length)
```

Comment:
It's not really a `endIndexInPsiFile`. Can we name it better? Like `boundedCursorPosition` or something along those lines.